test/extensions/network/dns_resolver/cares/dns_impl_test.cc

```diff
@@ -1340,26 +1448,42 @@ TEST_P(DnsImplTest, PendingTimerEnable) {
   Event::MockDispatcher dispatcher;
   Event::MockTimer* timer = new NiceMock<Event::MockTimer>();
   EXPECT_CALL(dispatcher, createTimer_(_)).WillOnce(Return(timer));
-  resolver_ = std::make_shared<DnsResolverImpl>(config, dispatcher, vec);
+  resolver_ = std::make_shared<DnsResolverImpl>(config, dispatcher, vec, stats_store_);
   Event::FileEvent* file_event = new NiceMock<Event::MockFileEvent>();
   EXPECT_CALL(dispatcher, createFileEvent_(_, _, _, _)).WillOnce(Return(file_event));
   EXPECT_CALL(*timer, enableTimer(_, _));
   EXPECT_NE(nullptr, resolveWithUnreferencedParameters("some.bad.domain.invalid",
                                                        DnsLookupFamily::V4Only, true));
+  checkStats(0 /*resolve_total*/, 1 /*pending_resolutions*/, 0 /*not_found*/,
```

Comment:
Probably worth adding a test to track >1 pending resolution and satisfactory popping of the resolutoins back to zero.